### ISSUE TITLE
VPA object should have condition to signalize no matching pods 

### ISSUE THREAD

**Issue by bskiba** [kubernetes/autoscaler#1181]
When the label selector on VPA doesn't match any pods in the cluster, the recomender should set a condition on VPA object to indicate this as it will cause us not to provide recommendations.

**Comment by phsiao** [comment]
@bskiba Happy to take this one.  Would something like

```
  status:
    conditions:
    - lastTransitionTime: 2018-08-30T23:14:13Z
      status: "False"
      reason: "NoMatchingPod"
      type: RecommendationProvided
```

be what you are looking for?   Or a separate condition?

**Comment by bskiba** [comment]
I think this looks very reasonable. 👍

**Comment by phsiao** [comment]
Right now, the status does not contain a conditions map like below if there is no matching pod

```
  status:
    recommendation: {}
```

it is easy to fix that by also setting things when `len(vpa.Recommendation.ContainerRecommendations) == 0`

to become

```
  status:
    conditions:
    - lastTransitionTime: 2018-08-31T19:34:44Z
      reason: NoMatchingPod
      status: "False"
      type: RecommendationProvided
    recommendation: {}
```

and would become normal again when pods are added to match

```
  status:
    conditions:
    - lastTransitionTime: 2018-08-31T19:40:01Z
      status: "True"
      type: RecommendationProvided
    recommendation:
      containerRecommendations:
      - containerName: hamster
        lowerBound:
          cpu: 190m
          memory: "104857600"
        target:
          cpu: 564m
          memory: "104857600"
        upperBound:
          cpu: 406644m
          memory: 4145750k
```

I still need to test for a couple of other cases and see how to address them, specifically,

1. when there are matching pods, but the pods are in crashloop/error/error pulling image so no stats to report, recommender would still report there are matching pods but no recommendations
2. when the pods are deleted, because of the internal model and vpa, it will continue to recommend stale data.

**Comment by phsiao** [comment]
I did run into a weird case that I have not had a chance to look into while working on this.

When I created a VPA first without matching pods, then add pods later, recommender does not make recommendations for a long time as if the pods don't exist.  Restarting recommender fixes that.

**Comment by bskiba** [comment]
Hm, I'm testing this right now from the current master branch, and I don't see this. I first created VPA pointing at a non-existing deployment. No recommendation was generated. Then created the deployment. After ~2 minutes got the recommendation.

**Comment by phsiao** [comment]
I can't reproduce it consistently yet.  I was going to try delete vpa and deployment independently several times to test, because that is what I was doing when I am testing a pr

**Comment by bskiba** [comment]
This has been fixed by #1915
